### ISSUE TITLE
bug: DGS plugin thinks my @InputArgument enum params mismatch, but they don't. 

### ISSUE THREAD

**Issue by NfNitLoop** [Netflix/dgs-framework#936]
## Expected behavior

DGS's IntelliJ correctly matches my query method signature to the corresponding types in my schema.graphqls.  (i.e.: there should be no warning for this case.)

## Actual behavior

The IntelliJ plugin tells me I'm using the wrong type.  It for some reason thinks it should be a collection type.

<img width="569" alt="Screen Shot 2022-03-16 at 2 04 06 PM" src="https://user-images.githubusercontent.com/339075/158690635-a02f389d-0557-4435-993c-8885b3874f70.png">
<img width="978" alt="Screen Shot 2022-03-16 at 2 04 35 PM" src="https://user-images.githubusercontent.com/339075/158690680-8b5b8418-0593-4109-ba65-ccbe10067d87.png">

Also, the suggested fix doesn't result in valid code:

<img width="848" alt="Screen Shot 2022-03-16 at 2 06 41 PM" src="https://user-images.githubusercontent.com/339075/158690991-c7ff8b10-a97f-4234-b7ec-f01b9a2f9448.png">

## Steps to reproduce

Create a new project with Spring Initializr: <https://start.spring.io/>

Select: 
* Gradle Project
* Kotlin (I'm not sure whether this affects Java too.)

Add DGS dependencies to the build.gradle.kts:

```kotlin
implementation(platform("com.netflix.graphql.dgs:graphql-dgs-platform-dependencies:latest.release"))
implementation("com.netflix.graphql.dgs:graphql-dgs-spring-boot-starter")
```

Add the DGS code-gen plugin for Gradle:
```kotlin
plugins {
    // [...]
    id("com.netflix.dgs.codegen") version "5.1.17"
}
```

Create src/main/resources/schema/schema.graphqls:

```graphqls
type Query {
    echoBool(myBool: MyBool): MyBool
}

enum MyBool {
    FOO
    BAR
}
```

Create the DGS component to implement Query:

```kotlin
@DgsComponent
class Query {
	@DgsQuery
	fun echoBool(@InputArgument myBool: MyBool?): MyBool? {
		return myBool
	}
}
```

Save and build, and you'll see the above errors in IntelliJ.

**Comment by srinivasankavitha** [comment]
Thanks for reporting, will look into this issue.

**Comment by srinivasankavitha** [comment]
Fixed in upcoming release `1.2.21` which will be available in a week or so.
